src/test/test_basemodel.py

```diff
@@ -0,0 +1,289 @@
+        except:
+            self.fail('Error: first.last@colorado.edu should be a valid email')
+        try:
+            BaseModel().is_valid_email('someUserName@someEmailDomain.something')
+        except:
+            self.fail('Error: this has the correct syntax for a valid email')
+        try:
+            BaseModel().is_valid_email('user.user.user123456789@colorado.edu')
+        except:
+            self.fail('Error: this has the correct syntax for a valid email')
+        with self.assertRaises(AssertionError):
+            BaseModel().is_valid_email('thisEmailShouldNotWork!@colorado.edu')
+        with self.assertRaises(AssertionError):
+            BaseModel().is_valid_email('invalidEmail@yahoo!.com')
+        #data_types = ['hello', 1, 1.5, True, [1,2,3], ['a','b','c']]
+        #with self.assertRaises(AssertionError):
+        #    for n in data_types:
+        #        BaseModel().is_valid_email(n)
+        with self.assertRaises(AssertionError):
+            BaseModel().is_valid_email('hello')
+        with self.assertRaises(Exception):
+           BaseModel().is_valid_email(1)
+        with self.assertRaises(Exception):
+           BaseModel().is_valid_email(1.5)
+        with self.assertRaises(Exception):
+           BaseModel().is_valid_email(True)
+        with self.assertRaises(Exception):
+           BaseModel().is_valid_email([1,2,3])
+        with self.assertRaises(Exception):
+           BaseModel().is_valid_email(['a','b','c'])
+
+    def test_is_list(self):
+        try:
+            BaseModel().is_list([1,2,3])
+        except:
+            self.fail('Error: [1,2,3] should be a valid list')
+        try:
+            BaseModel().is_list(['a'])
+        except:
+            self.fail('Error: this should be a valid list of one string entry')
+        try:
+            BaseModel().is_list(['hello', 1, 2.5, False, [1,2,3]])
+        except:
+            self.fail('Error: a list can consist of multiple data types')
+        try:
+            BaseModel().is_list(('hi', 1, True))
+        except:
+            self.fail('Error: tuples should be valid')
+        with self.assertRaises(AssertionError):
+            BaseModel().is_list(1)
+        with self.assertRaises(AssertionError):
+            BaseModel().is_list('hi')
+        with self.assertRaises(AssertionError):
+            BaseModel().is_list(True)
+
+    def test_is_none(self):
+        try:
+            BaseModel().is_none(None)
+        except:
+            self.fail('Error: None should be empty') 
+        with self.assertRaises(AssertionError):
+            BaseModel().is_none('')
+        with self.assertRaises(AssertionError):
+            BaseModel().is_none(0)
+        with self.assertRaises(AssertionError):
+            BaseModel().is_none([])
+        with self.assertRaises(AssertionError):
+            BaseModel().is_none(True)
+
+    def test_is_user(self):
+        # look into this method, its giving me error 'db is note defined'
```

Comment:
So i don't think there is such a thing as "db.exists()" (I am referring to the code in basemodel.py). I looked into rethinkdb's contains(), and I'm not sure if this is what you were wanting.